airbyte-server/src/main/java/io/airbyte/server/apis/HealthApiController.java

```diff
@@ -7,16 +7,23 @@
 import io.airbyte.api.generated.HealthApi;
 import io.airbyte.api.model.generated.HealthCheckRead;
 import io.airbyte.server.handlers.HealthCheckHandler;
-import javax.ws.rs.Path;
-import lombok.AllArgsConstructor;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import jakarta.inject.Singleton;
 
-@Path("/v1/health")
-@AllArgsConstructor
+@Controller("/api/v1/health")
+@Singleton
```

Comment:
You don't need both `@Singleton` and `@Controller`.  `@Controller` acts as both a singleton designation + the configuration bits to wire up the endpoint with the built-in Micronaut web server.